assets/js/googlesitekit/datastore/user/authentication.test.js

```diff
@@ -0,0 +1,334 @@
+				expect( isAuthenticated ).toEqual( responseAuthenticated.authenticated );
+			} );
+
+			it( 'dispatches an error if the request fails', async () => {
+				const response = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+				fetch
+					.doMockOnceIf( apiEndpoint )
+					.mockResponseOnce(
+						JSON.stringify( response ),
+						{ status: 500 }
+					);
+
+				muteConsole( 'error' );
+				registry.select( STORE_NAME ).isAuthenticated();
+				await subscribeUntil( registry,
+					// TODO: We may want a selector for this, but for now this is fine
+					// because it's internal-only.
+					() => store.getState().isFetchingAuthentication === false,
+				);
+
+				const isAuthenticated = registry.select( STORE_NAME ).isAuthenticated();
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( isAuthenticated ).toEqual( undefined );
+			} );
+
+			it( 'returns undefined if authentication info is not available', async () => {
+				// This triggers a network request, so ignore the error.
+				muteConsole( 'error' );
+				const isAuthenticated = registry.select( STORE_NAME ).isAuthenticated();
+
+				expect( isAuthenticated ).toEqual( undefined );
+			} );
+		} );
+
+		describe( 'getGrantedScopes', () => {
+			it( 'uses a resolver get all authentication info', async () => {
+				fetch
+					.doMockOnceIf( apiEndpoint )
+					.mockResponseOnce(
+						JSON.stringify( responseAuthenticated ),
+						{ status: 200 }
+					);
+
+				const initialIsAuthenticated = registry.select( STORE_NAME ).getGrantedScopes();
+				// The autentication info will be its initial value while the authentication
+				// info is fetched.
+				expect( initialIsAuthenticated ).toEqual( undefined );
+				await subscribeUntil( registry,
+					() => (
+						registry.select( STORE_NAME ).getGrantedScopes() !== undefined
+					),
+				);
+
+				const grantedScopes = registry.select( STORE_NAME ).getGrantedScopes();
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( grantedScopes ).toEqual( responseAuthenticated.grantedScopes );
+			} );
+
+			it( 'dispatches an error if the request fails', async () => {
+				const response = {
+					code: 'internal_server_error',
+					message: 'Internal server error',
+					data: { status: 500 },
+				};
+				fetch
+					.doMockOnceIf( apiEndpoint )
+					.mockResponseOnce(
+						JSON.stringify( response ),
+						{ status: 500 }
+					);
+
+				muteConsole( 'error' );
+				registry.select( STORE_NAME ).getGrantedScopes();
+				await subscribeUntil( registry,
+					// TODO: We may want a selector for this, but for now this is fine
+					// because it's internal-only.
+					() => store.getState().isFetchingAuthentication === false,
+				);
+
+				const grantedScopes = registry.select( STORE_NAME ).getGrantedScopes();
+
+				expect( fetch ).toHaveBeenCalledTimes( 1 );
+				expect( grantedScopes ).toEqual( undefined );
```

Comment:
Missing expectation about error 😄 (there may be other places that need this in other older tests too)